capsules/src/usb_user.rs

```diff
@@ -0,0 +1,137 @@
+//! USB system call interface
```

Comment:
Perhaps this (and the associated user-land library and app) can wait? I think if this PR implements a basic set of features for the SAM4L, and has an initial HIL that's already a fair bit. Along with the `usb_client` capsule, I think that'll be good.